src/client/csfle/options.rs

```diff
@@ -0,0 +1,140 @@
+use std::collections::HashMap;
+
+use bson::Array;
+use mongocrypt::ctx::KmsProvider;
+use typed_builder::TypedBuilder;
+
+use crate::{
+    bson::{doc, Bson, Document},
+    client::options::TlsOptions,
+    error::{Error, Result},
+};
+
+/// Options related to automatic encryption.
+///
+/// Automatic encryption is an enterprise only feature that only applies to operations on a
+/// collection. Automatic encryption is not supported for operations on a database or view, and
+/// operations that are not bypassed will result in error (see [libmongocrypt: Auto Encryption
+/// Allow-List](
+/// https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/client-side-encryption.rst#libmongocrypt-auto-encryption-allow-list
+/// )). To bypass automatic encryption for all operations, set bypassAutoEncryption=true in
+/// AutoEncryptionOpts.
+#[derive(Debug, Default, TypedBuilder, Clone)]
+#[non_exhaustive]
+#[builder(field_defaults(setter(into)))]
+pub struct AutoEncryptionOpts {
```

Comment:
nit: seems more consistent with our other options types to use the full `Options` rather than just `Opts`?